src/js/structs/DSLExpression.js

```diff
@@ -0,0 +1,63 @@
+import SearchDSL from '../../resources/grammar/SearchDSL.jison';
+
+/**
+ * This is a high-level compnent that wraps
+ */
+class DSLExpression {
+
+  /**
+   * Construct an immutable DSL expression
+   *
+   * @param {String} value - The expression string
+   */
+  constructor(value='') {
+    let ast = null;
+    let filter = (filters, resultset) => resultset;
+    let hasErrors = false;
+
+    if (value) {
+      // Try to parse the expression
+      try {
+        let result = SearchDSL.parse(value);
+        if (result.ast && result.filter) {
```

Comment:
Do we need to have this in `try` as well? Let's keep `try` as shallow as possible